src/github.com/matrix-org/dendrite/roomserver/input/events.go

```diff
@@ -44,3 +58,222 @@ func processRoomEvent(db RoomEventDatabase, input api.InputRoomEvent) error {
 	//      - The changes to the current state of the room.
 	panic("Not implemented")
 }
+
+// checkAuthEvents checks that the event passes authentication checks
+// Returns the numeric IDs for the auth events.
+func checkAuthEvents(db RoomEventDatabase, event gomatrixserverlib.Event, authEventIDs []string) ([]int64, error) {
+	// Grab the numeric IDs for the supplied auth state events from the database.
+	authStateEntries, err := db.StateEntriesForEventIDs(authEventIDs)
+	if err != nil {
+		return nil, err
+	}
+	if len(authStateEntries) < len(authEventIDs) {
+		return nil, fmt.Errorf("input: Some of the auth event IDs were missing from the database")
```

Comment:
It feels like this check should be the responsibility of `StateEntriesForEventIDs`. Why don't we just return an `err` if the output slice != input slice?